packages/auth0_api_python/tests/test_api_client.py

```diff
@@ -388,3 +404,1153 @@ async def test_verify_access_token_fail_no_audience_config():
+    """
+    access_token = "test_token"
+    dpop_proof = await generate_dpop_proof(
+        access_token=access_token,
+        http_method="GET",
+        http_url="https://api.example.com/resource"
+    )
+
+    api_client = ApiClient(
+        ApiClientOptions(domain="auth0.local", audience="my-audience")
+    )
+
+    # Verify the DPoP proof
+    claims = await api_client.verify_dpop_proof(
+        access_token=access_token,
+        proof=dpop_proof,
+        http_method="GET",
+        http_url="https://api.example.com/resource"
+    )
+    assert claims["jti"]  # Verify it has the required jti claim
+
+# --- Header Validation Tests ---
+
+@pytest.mark.asyncio
+async def test_verify_dpop_proof_fail_no_access_token():
+    """
+    Test that verify_dpop_proof fails when access_token is missing.
+    """
+    dpop_proof = await generate_dpop_proof(
+        access_token="test_token",
+        http_method="GET",
+        http_url="https://api.example.com/resource"
+    )
+
+    api_client = ApiClient(
+        ApiClientOptions(domain="auth0.local", audience="my-audience")
+    )
+
+    with pytest.raises(MissingRequiredArgumentError) as err:
+        await api_client.verify_dpop_proof(
+            access_token="",  # Empty access token
+            proof=dpop_proof,
+            http_method="GET",
+            http_url="https://api.example.com/resource"
+        )
+
+    assert "access_token" in str(err.value).lower()
+
+@pytest.mark.asyncio
+async def test_verify_dpop_proof_fail_no_dpop_proof():
+    """
+    Test that verify_dpop_proof fails when dpop_proof is missing.
+    """
+    api_client = ApiClient(
+        ApiClientOptions(domain="auth0.local", audience="my-audience")
+    )
+
+    with pytest.raises(MissingRequiredArgumentError) as err:
+        await api_client.verify_dpop_proof(
+            access_token="test_token",
+            proof="",  # Empty proof
+            http_method="GET",
+            http_url="https://api.example.com/resource"
+        )
+
+    assert "dpop_proof" in str(err.value).lower()
+
+@pytest.mark.asyncio
+async def test_verify_dpop_proof_fail_no_http_method_url():
+    """
+    Test that verify_dpop_proof fails when http_method or http_url is missing.
+    """
+    access_token = "test_token"
+    dpop_proof = await generate_dpop_proof(
+        access_token=access_token,
+        http_method="GET",
+        http_url="https://api.example.com/resource"
+    )
+
+    api_client = ApiClient(
+        ApiClientOptions(domain="auth0.local", audience="my-audience")
+    )
+
+    with pytest.raises(MissingRequiredArgumentError) as err:
+        await api_client.verify_dpop_proof(
+            access_token=access_token,
+            proof=dpop_proof,
+            http_method="",  # Empty method
+            http_url="https://api.example.com/resource"
+        )
+
+    assert "http_method" in str(err.value).lower() or "http_url" in str(err.value).lower()
```

Comment:
This should only assert `http_method`. `http_url` should be asserted in a separate test case.